gui-agent/vmside.c

```diff
@@ -1505,54 +1504,74 @@ static void mkghandles(Ghandles * g)
     g->screen = DefaultScreen(g->display);	/* get CRT id number */
     g->root_win = RootWindow(g->display, g->screen);	/* get default attributes */
     g->context = XCreateGC(g->display, g->root_win, 0, NULL);
-    g->wmDeleteMessage = XInternAtom(g->display, "WM_DELETE_WINDOW", False);
-    g->wmProtocols = XInternAtom(g->display, "WM_PROTOCOLS", False);
-    g->wm_hints = XInternAtom(g->display, "WM_HINTS", False);
-    g->wm_class = XInternAtom(g->display, "WM_CLASS", False);
-    g->utf8_string_atom = XInternAtom(g->display, "UTF8_STRING", False);
     g->stub_win = XCreateSimpleWindow(g->display, g->root_win,
             0, 0, 1, 1,
             0, BlackPixel(g->display,
                 g->screen),
             WhitePixel(g->display,
                 g->screen));
+    if ((unsigned)snprintf(tray_sel_atom_name, sizeof(tray_sel_atom_name),
+            "_NET_SYSTEM_TRAY_S%u", DefaultScreen(g->display)) >=
+        sizeof tray_sel_atom_name)
+        abort();
+#define SUPPORTED_ATOMS 6
+    const struct {
+        Atom *const dest;
+        const char *const name;
+    } atoms_to_intern[] = {
+        { &g->wmDeleteMessage,  "WM_DELETE_MESSAGE" },
+        { &g->wmProtocols,      "WM_PROTOCOLS" },
+        { &g->wm_hints,         "WM_HINTS" },
+        { &g->wm_class,         "WM_CLASS" },
+        { &g->tray_selection,   tray_sel_atom_name },
+        { &g->tray_opcode,      "_NET_SYSTEM_TRAY_MESSAGE_OPCODE" },
```

Comment:
`_NET_SYSTEM_TRAY_OPCODE`

Reply:
Fixed in 65d3025

Reply:
Strange, application icon appears (telegram-desktop), but nm-applet still not. Will test more

Reply:
https://github.com/QubesOS/qubes-core-agent-linux/blob/master/network/show-hide-nm-applet.sh ?

Reply:
Doesn't help. gsettings is true, 
```killall nm-applet && nm-applet &```
doesn't help too.